src/test/java/com/keboola/tableexporter/TableDataTest.java

```diff
@@ -0,0 +1,82 @@
+package com.keboola.tableexporter;
+
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.Arrays;
+import java.util.LinkedHashMap;
+
+import static org.mockito.Mockito.*;
+
+
+@RunWith(Parameterized.class)
+public class TableDataTest extends BaseTest {
+
+    private ResultSet resultSetMock;
+    private LinkedHashMap expectedOutput;
+
+    public TableDataTest(ResultSet resultSetMock, LinkedHashMap expectedOutput) {
+        this.resultSetMock = resultSetMock;
+        this.expectedOutput = expectedOutput;
+    }
+
+    @Parameterized.Parameters
+    public static Iterable<Object[]> data() throws SQLException {
+
+        return Arrays.asList(new Object[][] {
+                { getResultSetMock("normalTest"), getExpectedOutput("normalTest") },
+                { getResultSetMock("noRowCountTest"), getExpectedOutput("noRowCountTest") }
+        });
+    }
+
+    @Test
+    public void testTableData() throws SQLException {
+        MetaFetcher mf = new MetaFetcher(null);
```

Comment:
the mock seems to me a bit unnecessary when you can easily pass real db connection here, but ok